patches/tModLoader/Terraria/ModLoader/Utilities/HjsonExtensions.cs

```diff
@@ -71,23 +72,42 @@ private static void WriteFancyHjsonValue(TextWriter tw, JsonValue value, int lev
 				}
 
 				if (commentedObject != null) {
+					bool skipFirst = !showBraces;
 					string kwl = GetComments(commentedObject.Comments, "");
+					JsonType? lastJsonType = null;
 
 					foreach (string key in commentedObject.Order.Concat(commentedObject.Keys).Distinct()) {
 						if (!jObject.ContainsKey(key))
 							continue;
 
 						var val = jObject[key];
 
+						if ((val.JsonType != lastJsonType && lastJsonType != null) || lastJsonType == JsonType.Object) {
+							NewLine(tw, 0);
+						}
+
+						if (!string.IsNullOrWhiteSpace(kwl))
+							kwl = (isRootObject ? "" : "\n") + new string('\t', level + (showBraces ? 1 : 0)) + kwl.TrimStart();
+
+						lastJsonType = val.JsonType;
+						
 						tw.Write(kwl);
 
-						NewLine(tw, level + 1);
+						NewLine(tw, level + (showBraces ? 1 : 0));
+
 						kwl = GetComments(commentedObject.Comments, key);
 
+						bool commentedOut = jObject is CommentedWscJsonObject commented && commented.CommentedOut.Contains(key);
```

Comment:
I don't love this approach, the code doesn't really go here, and the hjson object will not come back the same when deserialized again. 

It doesn't hurt because we only do it in generated files but...

Could we not generate the comments directly when we generate the files?

Reply:
Guess we should talk about this one. This is where we are generating the files, doing it after this method would require writing parsing code

Reply:
I wanted to generate actual comments instead, before the writer, but I guess you can't because you need the `jObject` to be formatted by the writer...

What does an ExampleMod output look like for a new language?